spinn_front_end_common/interface/interface_functions/router_provenance_gatherer.py

```diff
@@ -135,288 +180,236 @@ def _add_router_provenance_data(
             except:  # noqa: E722
                 logger.warning(
                     "Could not read routing diagnostics from {}, {}",
                     x, y, exc_info=True)
-
-    def _add_router_chip_diagnostic(
-            self, chip, seen_chips, items, reinjection_data):
+                return
+            seen_chips.add((x, y))
+            status = self.__get_status(reinjection_data, x, y)
+            items.extend(self.__router_diagnostics(
+                x, y, diagnostics, status, True, table))
+            self.__add_totals(diagnostics, status)
+
+    def _add_unseen_router_chip_diagnostic(
+            self, chip, items, reinjection_data):
         """
         :param ~.Chip chip:
         :param set(tuple(int,int)) seen_chips:
         :param list(ProvenanceDataItem) items:
         :param dict(tuple(int,int),ReInjectionStatus) reinjection_data:
         """
         # pylint: disable=bare-except
-        if not chip.virtual and (chip.x, chip.y) not in seen_chips:
-            try:
-                diagnostic = self._txrx.get_router_diagnostics(chip.x, chip.y)
-
-                if (diagnostic.n_dropped_multicast_packets or
-                        diagnostic.n_local_multicast_packets or
-                        diagnostic.n_external_multicast_packets):
-                    reinjection_status = None
-                    if reinjection_data is not None:
-                        reinjection_status = reinjection_data[chip.x, chip.y]
-                    self.__add_router_diagnostics(
-                        items, chip.x, chip.y, diagnostic, reinjection_status,
-                        False, None)
-                    self._add_totals(diagnostic, reinjection_status)
-            except:  # noqa: E722
-                # There could be issues with unused chips - don't worry!
-                pass
+        try:
+            diagnostics = self._txrx.get_router_diagnostics(chip.x, chip.y)
+        except:  # noqa: E722
+            # There could be issues with unused chips - don't worry!
+            return
+        if (diagnostics.n_dropped_multicast_packets or
+                diagnostics.n_local_multicast_packets or
+                diagnostics.n_external_multicast_packets):
+            status = self.__get_status(reinjection_data, chip.x, chip.y)
+            items.extend(self.__router_diagnostics(
+                chip.x, chip.y, diagnostics, status, False, None))
+            self.__add_totals(diagnostics, status)
 
-    def _add_totals(self, router_diagnostic, reinjection_status):
+    @staticmethod
+    def __get_status(reinjection_data, x, y):
         """
-        :param ~.RouterDiagnostics router_diagnostic:
-        :param ReInjectionStatus reinjection_status:
+        :param dict(tuple(int,int),ReInjectionStatus) reinjection_data:
+        :param int x:
+        :param int y:
+        :rtype: ReInjectionStatus or None
+        """
+        status = None
+        if reinjection_data is not None:
```

Comment:
why have a temp status instead of just two returns?

Reply:
No idea what I was thinking there.